src/components/Checkbox/Checkbox.tsx

```diff
@@ -303,7 +310,11 @@ const inverseHasErrorBoxVariants = {
 export type CheckboxClasses = GetClasses<typeof useStyles>;
 
 export type CheckboxProps = BaseFormElementWithNodeLabel &
-  ClassOverrides<CheckboxClasses> & { indeterminate?: boolean };
+  ClassOverrides<CheckboxClasses> & {
+    indeterminate?: boolean;
+    /** This property shows the required asterisk (*). Required validation needs to be implemented separately. */
+    showRequiredLabel?: boolean;
+  };
```

Comment:
Would it make sense to add the `showRequiredLabel` prop to `BaseFormElementWithNodeLabel` to avoid duplication?

Reply:
Yeah that makes sense to me. I notice Checkbox and Toggle are the only two that use `BaseFormElementWithNodeLabel` - is there a reason for this? If not, would it be helpful to refactor all components to extend `BaseFormElementWithNodeLabel` and remove redundant props?

Reply:
Yeah, probably.

Reply:
Made a [ticket for it on the Chroma board](https://lifeomic.atlassian.net/browse/CHROM-203). Haley and I will prioritize it after we finish the upgrade to MUI 5